src/assignments/yunfeng/lesson_4_answers.clj

```diff
@@ -0,0 +1,138 @@
+;;recogize a card problem
+(def suit-map {:S :spade :H :heart :D :diamond :C :club})
+(def rank-map {:2 0 :3 1 :4 2 :5 3 :6 4 :7 5 :8 6 :9 7 :T 8 :J 9 :Q 10 :K 11 :A 12})
+
+(defn parse-suit
+  "parse the string to get the suit and return it as a keyword"
+  [card-string]
+  (keyword (subs card-string 0 1)))
+
+;;test parse-suit function
+(parse-suit "ST")
+
+(defn parse-rank
+  "parse the string to get the rank and return it as a keyword"
+  [card-string]
+  (keyword (subs card-string 1)))
```

Comment:
Interesting idea to use keywords throughout.  Can you explain why you chose this, as opposed to, say, just characters?

Reply:
I want to use keyword because it is can be used as a function while characters can not.